caiman/utils/visualization.py

```diff
@@ -1217,3 +1218,143 @@ def update(val):
+        patch_rows: ndarray
+            num_patch_rows x 2 array, where row i contains onset and offset row pixels for patch row i
+        patch_cols: ndarray
+            num_patch_cols x 2 array, where row j contains onset and offset column pixels for patch column j
+            
+    Note: 
+        Currently assumes square patches so takes in a single number for stride/overlap. 
+    """
+    patch_width = overlap + stride
+    
+    patch_onset_rows = np.array(list(range(0, im_dims[0] - patch_width, stride)) + [im_dims[0] - patch_width])
+    patch_offset_rows = patch_onset_rows + patch_width
+    patch_offset_rows[patch_offset_rows > im_dims[0]-1] = im_dims[0]-1
+    patch_rows = np.column_stack((patch_onset_rows, patch_offset_rows))
+    
+    patch_onset_cols = np.array(list(range(0, im_dims[1] - patch_width, stride)) + [im_dims[1] - patch_width])
+    patch_offset_cols = patch_onset_cols + patch_width 
+    patch_offset_cols[patch_offset_cols > im_dims[1]-1] = im_dims[1]-1
+    patch_cols = np.column_stack((patch_onset_cols, patch_offset_cols))
+    
+    return patch_rows, patch_cols
+
+
+def rect_draw(row_minmax: ArrayLike, 
+              col_minmax: ArrayLike, 
+              color: Optional[str]='white', 
+              alpha: Optional[float]=0.3, 
+              ax: Optional[Any]=None) -> Tuple[Any, Any]:
+    """
+    Draw a single transluscent rectangle on given axes object.
+    
+    Args:
+        row_minmax: array-like
+            [row_min, row_max] -- 2-elt int bounds for rect rows
+        col_minmax: array-like 
+            [col_min, col_max] -- int bounds for rect cols
+        color : string
+            rectangle color, default 'yellow'
+        alpha : float
+            rectangle alpha (0. to 1., where 1 is opaque), default 0.3
+        ax : pyplot.Axes object
+            axes object upon which rectangle will be drawn, default None
+    
+    Returns:
+        ax (Axes object)
+        rect (Rectangle object)
+    """
+    from matplotlib.patches import Rectangle
+    
+    if ax is None:
+        ax = pl.gca()
+        
+    box_origin = (col_minmax[0], row_minmax[0])
+    box_height = row_minmax[1] - row_minmax[0] 
+    box_width = col_minmax[1] - col_minmax[0]
+
+    rect = Rectangle(box_origin, 
+                     width=box_width, 
+                     height=box_height,
+                     color=color, 
+                     alpha=alpha)
+    ax.add_patch(rect)
+
+    return ax, rect
+
+
+def view_quilt(template_image: np.ndarray, 
+               stride: int, 
+               overlap: int, 
+               color: Optional[str]='white', 
+               alpha: Optional[float]=0.3, 
+               vmin: Optional[float]=0.0, 
+               vmax: Optional[float]=0.6, 
+               figsize: Optional[Tuple]=(6,6)) -> Any:
+    """
+    Plot patches on template image given stride and overlap parameters. 
+    
+    Given stride and overlap, plot overlapping patch blocks on template image.
```

Comment:
Seems to have some redundancy between the 1-sentence and longer description; is this because some tool treats those as separate?

Reply:
Yes this is sort of silly. I can just combine them.